File: app/src/main/java/br/com/vicentec12/mygames/ui/console/ConsoleHolder.java
package br.com.vicentec12.mygames.ui.console;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import br.com.vicentec12.mygames.data.model.ConsoleWithGames;
import br.com.vicentec12.mygames.databinding.ItemConsoleBinding;
import br.com.vicentec12.mygames.interfaces.OnItemClickListener;

public class ConsoleHolder extends RecyclerView.ViewHolder {

    private ItemConsoleBinding mBinding;
    private ConsoleWithGames mConsoleWithGames;
    private OnItemClickListener mOnItemClickListener;

    public ConsoleHolder(@NonNull ItemConsoleBinding mBinding, OnItemClickListener mOnItemClickListener) {
        super(mBinding.getRoot());
        this.mBinding = mBinding;
        this.mOnItemClickListener = mOnItemClickListener;
    }

    public void bind(ConsoleWithGames consoleWithGames) {
        this.mConsoleWithGames = consoleWithGames;
        mBinding.setPosition(getAdapterPosition());
        mBinding.setConsoleWithGames(mConsoleWithGames);
        mBinding.setOnItemClickListener(mOnItemClickListener);
    }

}

File: app/src/main/java/br/com/vicentec12/mygames/ui/console/ConsoleHolder.kt
package br.com.vicentec12.mygames.ui.console

import androidx.recyclerview.widget.RecyclerView
import br.com.vicentec12.mygames.data.model.ConsoleWithGames
import br.com.vicentec12.mygames.databinding.ItemConsoleBinding
import br.com.vicentec12.mygames.interfaces.OnItemClickListener

class ConsoleHolder(
        private val mBinding: ItemConsoleBinding,
        private val mOnItemClickListener: OnItemClickListener?
) : RecyclerView.ViewHolder(mBinding.root) {

    fun bind(mConsoleWithGames: ConsoleWithGames) {
        mBinding.position = adapterPosition
        mBinding.consoleWithGames = mConsoleWithGames
        mBinding.onItemClickListener = mOnItemClickListener
    }

}